include/oneapi/dpl/pstl/glue_memory_impl.h

```diff
@@ -262,9 +271,18 @@ destroy_n(_ExecutionPolicy&& __exec, _ForwardIterator __first, _Size __n)
     else
     {
 #if _ONEDPL_ICPX_OMP_SIMD_DESTROY_WINDWOS_BROKEN
-        return oneapi::dpl::__internal::__pattern_walk1_n(::std::forward<_ExecutionPolicy>(__exec), __first, __n,
-                                                          [](_ReferenceType __val) { __val.~_ValueType(); },
-                                                          /*__is_vector*/ ::std::false_type{}, __is_parallel);
+        if constexpr (oneapi::dpl::__internal::__enable_if_host_execution_policy<_ExecutionPolicy>::value)
```

Comment:
The same

Reply:
It's been implemented as a part of https://github.com/oneapi-src/oneDPL/pull/1182#discussion_r1328439292